<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Plotting &mdash; scCCA 0.1.0 documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="_static/jquery.js"></script>
        <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/doctools.js"></script>
        <script src="_static/sphinx_highlight.js"></script>
        <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Analysing PBMC5K using scCCA" href="notebooks/PBMC5K.html" />
    <link rel="prev" title="Models" href="models.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            scCCA
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">API:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="models.html">Models</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Plotting</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#module-scCCA.plots.qc">Quality control</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#scCCA.plots.qc.disp"><code class="docutils literal notranslate"><span class="pre">disp()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#scCCA.plots.qc.qc_hist"><code class="docutils literal notranslate"><span class="pre">qc_hist()</span></code></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#module-scCCA.plots.loadings_scatter">Plotting loadings</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#scCCA.plots.loadings_scatter.StateData"><code class="docutils literal notranslate"><span class="pre">StateData</span></code></a><ul>
<li class="toctree-l4"><a class="reference internal" href="#scCCA.plots.loadings_scatter.StateData.c"><code class="docutils literal notranslate"><span class="pre">StateData.c</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#scCCA.plots.loadings_scatter.StateData.g"><code class="docutils literal notranslate"><span class="pre">StateData.g</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#scCCA.plots.loadings_scatter.StateData.go"><code class="docutils literal notranslate"><span class="pre">StateData.go</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#scCCA.plots.loadings_scatter.StateData.o"><code class="docutils literal notranslate"><span class="pre">StateData.o</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#scCCA.plots.loadings_scatter.StateData.sz"><code class="docutils literal notranslate"><span class="pre">StateData.sz</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#scCCA.plots.loadings_scatter.StateData.x"><code class="docutils literal notranslate"><span class="pre">StateData.x</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#scCCA.plots.loadings_scatter.StateData.xo"><code class="docutils literal notranslate"><span class="pre">StateData.xo</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#scCCA.plots.loadings_scatter.StateData.y"><code class="docutils literal notranslate"><span class="pre">StateData.y</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#scCCA.plots.loadings_scatter.StateData.yo"><code class="docutils literal notranslate"><span class="pre">StateData.yo</span></code></a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#scCCA.plots.loadings_scatter.loadings_scatter"><code class="docutils literal notranslate"><span class="pre">loadings_scatter()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#scCCA.plots.loadings_state.loadings_state"><code class="docutils literal notranslate"><span class="pre">loadings_state()</span></code></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#module-scCCA.plots.cluster_enrichment">Plotting factors</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#scCCA.plots.cluster_enrichment.cluster_enrichment"><code class="docutils literal notranslate"><span class="pre">cluster_enrichment()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#scCCA.plots.factor_embedding.factor_embedding"><code class="docutils literal notranslate"><span class="pre">factor_embedding()</span></code></a></li>
</ul>
</li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Tutorials:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="notebooks/PBMC5K.html">Analysing PBMC5K using scCCA</a></li>
<li class="toctree-l1"><a class="reference internal" href="notebooks/kang.html">Analysing interferon-<span class="math notranslate nohighlight">\(\beta\)</span> stimulated PBMCs using scPCA</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">scCCA</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Plotting</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/plots.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  
<style>
/* CSS overrides for sphinx_rtd_theme */

/* 24px margin */
.nbinput.nblast.container,
.nboutput.nblast.container {
    margin-bottom: 19px;  /* padding has already 5px */
}

/* ... except between code cells! */
.nblast.container + .nbinput.container {
    margin-top: -19px;
}

.admonition > p:before {
    margin-right: 4px;  /* make room for the exclamation icon */
}

/* Fix math alignment, see https://github.com/rtfd/sphinx_rtd_theme/pull/686 */
.math {
    text-align: unset;
}
</style>
<section id="plotting">
<h1>Plotting<a class="headerlink" href="#plotting" title="Permalink to this heading"></a></h1>
<section id="module-scCCA.plots.qc">
<span id="quality-control"></span><h2>Quality control<a class="headerlink" href="#module-scCCA.plots.qc" title="Permalink to this heading"></a></h2>
<dl class="py function">
<dt class="sig sig-object py" id="scCCA.plots.qc.disp">
<span class="sig-prename descclassname"><span class="pre">scCCA.plots.qc.</span></span><span class="sig-name descname"><span class="pre">disp</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">adata</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">model_key</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'scpca'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">layers_key</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">protein_obsm_key</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cmap</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'viridis'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ax</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#scCCA.plots.qc.disp" title="Permalink to this definition"></a></dt>
<dd><p>Plot the fitted dispersion against the coefficient of variation RNA or
protein counts.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>adata</strong> (<em>AnnData</em>) – AnnData object.</p></li>
<li><p><strong>model_key</strong> (<em>str</em><em>, </em><em>optional</em><em> (</em><em>default: &quot;scpca&quot;</em><em>)</em>) – Key for the fitted model.</p></li>
<li><p><strong>layers_key</strong> (<em>str</em><em>, </em><em>optional</em><em> (</em><em>default: None</em><em>)</em>) – If <cite>layers_key</cite> is None, then the raw counts are extracted from <cite>adata.X</cite>.
Otherwise, the counts are extracted from <cite>adata.layers[layers_key]</cite>.</p></li>
<li><p><strong>protein_obsm_key</strong> (<em>str</em><em>, </em><em>optional</em><em> (</em><em>default: None</em><em>)</em>) – Key for protein counts in <cite>adata.obsm</cite>. Providing <cite>protein_obsm_key</cite>
overrides <cite>layers_key</cite>, i.e. protein counts are plotted.</p></li>
<li><p><strong>cmap</strong> (<em>str</em><em>, </em><em>optional</em><em> (</em><em>default: &quot;viridis&quot;</em><em>)</em>) – Colormap for the scatterplot. Color represents the mean of the counts.</p></li>
<li><p><strong>ax</strong> (<em>matplotlib.axes.Axes</em><em>, </em><em>optional</em><em> (</em><em>default: None</em><em>)</em>) – Axes to plot on. If None, then a new figure is created.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>ax</strong></p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>matplotlib.axes.Axes</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="scCCA.plots.qc.qc_hist">
<span class="sig-prename descclassname"><span class="pre">scCCA.plots.qc.</span></span><span class="sig-name descname"><span class="pre">qc_hist</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">adata</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">model_key</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'scpca'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">layers_key</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">protein_obsm_key</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cmap</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'viridis'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">colorbar_pos</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'right'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">colorbar_width</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'3%'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">orientation</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'vertical'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ax</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#scCCA.plots.qc.qc_hist" title="Permalink to this definition"></a></dt>
<dd><p>Plots a 2D histogram of the predicted counts against the true counts.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>adata</strong> (<em>AnnData</em>) – AnnData object.</p></li>
<li><p><strong>model_key</strong> (<em>str</em><em>, </em><em>optional</em><em> (</em><em>default: &quot;scpca&quot;</em><em>)</em>) – Key for the fitted model.</p></li>
<li><p><strong>layers_key</strong> (<em>str</em><em>, </em><em>optional</em><em> (</em><em>default: None</em><em>)</em>) – If <cite>layers_key</cite> is None, then the raw counts are extracted from <cite>adata.X</cite>.
Otherwise, the counts are extracted from <cite>adata.layers[layers_key]</cite>.</p></li>
<li><p><strong>protein_obsm_key</strong> (<em>str</em><em>, </em><em>optional</em><em> (</em><em>default: None</em><em>)</em>) – Key for protein counts in <cite>adata.obsm</cite>. Providing <cite>protein_obsm_key</cite>
overrides <cite>layers_key</cite>, i.e. protein counts are plotted.</p></li>
<li><p><strong>cmap</strong> (<em>str</em><em>, </em><em>optional</em><em> (</em><em>default: &quot;viridis&quot;</em><em>)</em>) – Colormap for the scatterplot. Color represents the mean of the counts.</p></li>
<li><p><strong>ax</strong> (<em>matplotlib.axes.Axes</em><em>, </em><em>optional</em><em> (</em><em>default: None</em><em>)</em>) – Axes to plot on. If None, then a new figure is created.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>ax</strong></p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>matplotlib.axes.Axes</p>
</dd>
</dl>
</dd></dl>

</section>
<section id="module-scCCA.plots.loadings_scatter">
<span id="plotting-loadings"></span><h2>Plotting loadings<a class="headerlink" href="#module-scCCA.plots.loadings_scatter" title="Permalink to this heading"></a></h2>
<dl class="py class">
<dt class="sig sig-object py" id="scCCA.plots.loadings_scatter.StateData">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">scCCA.plots.loadings_scatter.</span></span><span class="sig-name descname"><span class="pre">StateData</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">x</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">y</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">c</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">o</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">g</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">xo</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">yo</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sz</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">go</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#scCCA.plots.loadings_scatter.StateData" title="Permalink to this definition"></a></dt>
<dd><dl class="py attribute">
<dt class="sig sig-object py" id="scCCA.plots.loadings_scatter.StateData.c">
<span class="sig-name descname"><span class="pre">c</span></span><a class="headerlink" href="#scCCA.plots.loadings_scatter.StateData.c" title="Permalink to this definition"></a></dt>
<dd><p>Alias for field number 2</p>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="scCCA.plots.loadings_scatter.StateData.g">
<span class="sig-name descname"><span class="pre">g</span></span><a class="headerlink" href="#scCCA.plots.loadings_scatter.StateData.g" title="Permalink to this definition"></a></dt>
<dd><p>Alias for field number 4</p>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="scCCA.plots.loadings_scatter.StateData.go">
<span class="sig-name descname"><span class="pre">go</span></span><a class="headerlink" href="#scCCA.plots.loadings_scatter.StateData.go" title="Permalink to this definition"></a></dt>
<dd><p>Alias for field number 8</p>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="scCCA.plots.loadings_scatter.StateData.o">
<span class="sig-name descname"><span class="pre">o</span></span><a class="headerlink" href="#scCCA.plots.loadings_scatter.StateData.o" title="Permalink to this definition"></a></dt>
<dd><p>Alias for field number 3</p>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="scCCA.plots.loadings_scatter.StateData.sz">
<span class="sig-name descname"><span class="pre">sz</span></span><a class="headerlink" href="#scCCA.plots.loadings_scatter.StateData.sz" title="Permalink to this definition"></a></dt>
<dd><p>Alias for field number 7</p>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="scCCA.plots.loadings_scatter.StateData.x">
<span class="sig-name descname"><span class="pre">x</span></span><a class="headerlink" href="#scCCA.plots.loadings_scatter.StateData.x" title="Permalink to this definition"></a></dt>
<dd><p>Alias for field number 0</p>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="scCCA.plots.loadings_scatter.StateData.xo">
<span class="sig-name descname"><span class="pre">xo</span></span><a class="headerlink" href="#scCCA.plots.loadings_scatter.StateData.xo" title="Permalink to this definition"></a></dt>
<dd><p>Alias for field number 5</p>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="scCCA.plots.loadings_scatter.StateData.y">
<span class="sig-name descname"><span class="pre">y</span></span><a class="headerlink" href="#scCCA.plots.loadings_scatter.StateData.y" title="Permalink to this definition"></a></dt>
<dd><p>Alias for field number 1</p>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="scCCA.plots.loadings_scatter.StateData.yo">
<span class="sig-name descname"><span class="pre">yo</span></span><a class="headerlink" href="#scCCA.plots.loadings_scatter.StateData.yo" title="Permalink to this definition"></a></dt>
<dd><p>Alias for field number 6</p>
</dd></dl>

</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="scCCA.plots.loadings_scatter.loadings_scatter">
<span class="sig-prename descclassname"><span class="pre">scCCA.plots.loadings_scatter.</span></span><span class="sig-name descname"><span class="pre">loadings_scatter</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">adata</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">model_key</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">states=[]</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">factor=None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">genes=[]</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">diff=[]</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">geneset=None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">vector='W_rna'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">alpha=1.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">highest=3</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">lowest=3</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">threshold=None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">size_scale=1.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sign=1.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">jitter=0.01</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fontsize=10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">show_labels=0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">plot_diff=False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">return_order=False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">annotation_linewidth=0.5</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">show_geneset=False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">show_diff=False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">geneset_top_genes=100</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">geneset_bottom_genes=0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">organism='Human'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cmap=&lt;matplotlib.colors.LinearSegmentedColormap</span> <span class="pre">object&gt;</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">format_func=&lt;function</span> <span class="pre">&lt;lambda&gt;&gt;</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ncols=4</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">width=4</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">height=3</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ax=None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#scCCA.plots.loadings_scatter.loadings_scatter" title="Permalink to this definition"></a></dt>
<dd><p>Plot factor on a given embedding.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>adata</strong> (<em>AnnData</em>) – AnnData object.</p></li>
<li><p><strong>model_key</strong> (<em>str</em><em>, </em><em>optional</em><em> (</em><em>default: &quot;X_scpca&quot;</em><em>)</em>) – Key for the fitted model.</p></li>
<li><p><strong>embedding</strong> (<em>str</em><em>, </em><em>optional</em><em> (</em><em>default: &quot;X_umap&quot;</em><em>)</em>) – Key for the embedding (e.g. UMAP, T-SNE).</p></li>
<li><p><strong>factor</strong> (<em>int</em><em>, </em><em>list</em><em>, </em><em>optional</em><em> (</em><em>default: None</em><em>)</em>) – Factor(s) to plot. If None, then all factors are plotted.</p></li>
<li><p><strong>sign</strong> (<em>float</em><em>, </em><em>optional</em><em> (</em><em>default: 1.0</em><em>)</em>) – Sign of the factor. Should be either 1.0 or -1.0.</p></li>
<li><p><strong>cmap</strong> (<em>str</em><em>, </em><em>optional</em><em> (</em><em>default: &quot;PiYG&quot;</em><em>)</em>) – Colormap for the scatterplot.</p></li>
<li><p><strong>colorbar_pos</strong> (<em>str</em><em>, </em><em>optional</em><em> (</em><em>default: &quot;right&quot;</em><em>)</em>) – Position of the colorbar.</p></li>
<li><p><strong>colorbar_width</strong> (<em>str</em><em>, </em><em>optional</em><em> (</em><em>default: &quot;3%&quot;</em><em>)</em>) – Width of the colorbar.</p></li>
<li><p><strong>orientation</strong> (<em>str</em><em>, </em><em>optional</em><em> (</em><em>default: &quot;vertical&quot;</em><em>)</em>) – Orientation of the colorbar. Should be either “vertical” or “horizontal”.</p></li>
<li><p><strong>size</strong> (<em>float</em><em>, </em><em>optional</em><em> (</em><em>default: 1</em><em>)</em>) – Marker/Dot size of the scatterplot.</p></li>
<li><p><strong>ncols</strong> (<em>int</em><em>, </em><em>optional</em><em> (</em><em>default: 4</em><em>)</em>) – Number of columns for the subplots.</p></li>
<li><p><strong>width</strong> (<em>int</em><em>, </em><em>optional</em><em> (</em><em>default: 4</em><em>)</em>) – Width of each subplot.</p></li>
<li><p><strong>height</strong> (<em>int</em><em>, </em><em>optional</em><em> (</em><em>default: 3</em><em>)</em>) – Height of each subplot.</p></li>
<li><p><strong>ax</strong> (<em>matplotlib.axes.Axes</em><em>, </em><em>optional</em><em> (</em><em>default: None</em><em>)</em>) – Axes object to plot on. If None, then a new figure is created.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>ax</strong> – Axes object.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>matplotlib.axes.Axes</p>
</dd>
</dl>
</dd></dl>

<span class="target" id="module-scCCA.plots.loadings_state"></span><dl class="py function">
<dt class="sig sig-object py" id="scCCA.plots.loadings_state.loadings_state">
<span class="sig-prename descclassname"><span class="pre">scCCA.plots.loadings_state.</span></span><span class="sig-name descname"><span class="pre">loadings_state</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">adata</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">model_key</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">states</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">factor=None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">variable='W_rna'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">highest=10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">lowest=10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">threshold=None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sign=1.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">geneset=None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">geneset_top_genes=100</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">geneset_bottom_genes=0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">organism='Human'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cmap=&lt;matplotlib.colors.LinearSegmentedColormap</span> <span class="pre">object&gt;</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">colorbar_pos='right'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">colorbar_width='3%'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">orientation='vertical'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fontsize=10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pad=0.1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">show_corr=False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">show_rank=False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">show_diff=False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">show_lines=False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">size_func=&lt;function</span> <span class="pre">&lt;lambda&gt;&gt;</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">text_func=&lt;function</span> <span class="pre">&lt;lambda&gt;&gt;</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sharey=False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sharex=False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ncols=4</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">width=4</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">height=3</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">text_kwargs={}</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ax=None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#scCCA.plots.loadings_state.loadings_state" title="Permalink to this definition"></a></dt>
<dd><p>Visualize the loading state of a given model on the data.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>adata</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">AnnData</span></code>) – The annotated data matrix.</p></li>
<li><p><strong>model_key</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">str</span></code>) – Key to access the model from the data.</p></li>
<li><p><strong>states</strong> (<code class="xref py py-data docutils literal notranslate"><span class="pre">Union</span></code>[<code class="xref py py-class docutils literal notranslate"><span class="pre">str</span></code>, <code class="xref py py-class docutils literal notranslate"><span class="pre">List</span></code>[<code class="xref py py-class docutils literal notranslate"><span class="pre">str</span></code>]]) – List of states to consider. Default is an empty list.</p></li>
<li><p><strong>factor</strong> (<code class="xref py py-data docutils literal notranslate"><span class="pre">Union</span></code>[<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>, <code class="xref py py-class docutils literal notranslate"><span class="pre">List</span></code>[<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>], <code class="xref py py-obj docutils literal notranslate"><span class="pre">None</span></code>]) – Factor or factors to consider. Default is None.</p></li>
<li><p><strong>variable</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">str</span></code>) – Variable to consider. Default is “W_rna”.</p></li>
<li><p><strong>highest</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>) – Number of highest values to consider. Default is 10.</p></li>
<li><p><strong>lowest</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>) – Number of lowest values to consider. Default is 10.</p></li>
<li><p><strong>threshold</strong> (<code class="xref py py-data docutils literal notranslate"><span class="pre">Optional</span></code>[<code class="xref py py-class docutils literal notranslate"><span class="pre">float</span></code>]) – Threshold value to consider. Default is None.</p></li>
<li><p><strong>sign</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>) – Sign value. Default is 1.0.</p></li>
<li><p><strong>geneset</strong> (<code class="xref py py-data docutils literal notranslate"><span class="pre">Optional</span></code>[<code class="xref py py-class docutils literal notranslate"><span class="pre">str</span></code>]) – Set of genes to consider. Default is None.</p></li>
<li><p><strong>geneset_top_genes</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>) – Number of top genes in the gene set to consider. Default is 100.</p></li>
<li><p><strong>geneset_bottom_genes</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>) – Number of bottom genes in the gene set to consider. Default is 0.</p></li>
<li><p><strong>organism</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">str</span></code>) – Organism to consider. Default is “Human”.</p></li>
<li><p><strong>cmap</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">Colormap</span></code>) – Colormap to use. Default is cm.RdBu.</p></li>
<li><p><strong>colorbar_pos</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">str</span></code>) – Position of the colorbar. Default is “right”.</p></li>
<li><p><strong>colorbar_width</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">str</span></code>) – Width of the colorbar. Default is “3%”.</p></li>
<li><p><strong>orientation</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">str</span></code>) – Orientation of the plot. Default is “vertical”.</p></li>
<li><p><strong>fontsize</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>) – Font size to use. Default is 10.</p></li>
<li><p><strong>pad</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">float</span></code>) – Padding value. Default is 0.1.</p></li>
<li><p><strong>show_corr</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>) – Whether to show correlation. Default is False.</p></li>
<li><p><strong>show_rank</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>) – Whether to show rank. Default is False.</p></li>
<li><p><strong>show_diff</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>) – Whether to show difference. Default is False.</p></li>
<li><p><strong>show_lines</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>) – Whether to show lines. Default is False.</p></li>
<li><p><strong>size_func</strong> (<code class="xref py py-data docutils literal notranslate"><span class="pre">Callable</span></code>) – Function to determine size. Default is a function that returns 10.</p></li>
<li><p><strong>text_func</strong> (<code class="xref py py-data docutils literal notranslate"><span class="pre">Callable</span></code>) – Function to format text. Default is a function that wraps text.</p></li>
<li><p><strong>sharey</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>) – Whether to share y-axis. Default is False.</p></li>
<li><p><strong>sharex</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>) – Whether to share x-axis. Default is False.</p></li>
<li><p><strong>ncols</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>) – Number of columns. Default is 4.</p></li>
<li><p><strong>width</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>) – Width of the plot. Default is 4.</p></li>
<li><p><strong>height</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>) – Height of the plot. Default is 3.</p></li>
<li><p><strong>text_kwargs</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">dict</span></code>) – Keyword arguments for text formatting. Default is an empty dictionary.</p></li>
<li><p><strong>ax</strong> (<code class="xref py py-data docutils literal notranslate"><span class="pre">Optional</span></code>[<code class="xref py py-class docutils literal notranslate"><span class="pre">Axes</span></code>]) – Axis object. Default is None.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>Configured axis object.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>ax</p>
</dd>
</dl>
<p class="rubric">Notes</p>
<p>This function visualizes the loading state of a given model on the data.
It provides various customization options for visualization.</p>
</dd></dl>

</section>
<section id="module-scCCA.plots.cluster_enrichment">
<span id="plotting-factors"></span><h2>Plotting factors<a class="headerlink" href="#module-scCCA.plots.cluster_enrichment" title="Permalink to this heading"></a></h2>
<dl class="py function">
<dt class="sig sig-object py" id="scCCA.plots.cluster_enrichment.cluster_enrichment">
<span class="sig-prename descclassname"><span class="pre">scCCA.plots.cluster_enrichment.</span></span><span class="sig-name descname"><span class="pre">cluster_enrichment</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">adata</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">model_key</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">states</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">factor</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">state_key</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cluster_key</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">groups</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dot</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sign</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">highest</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">lowest</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">threshold</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1.96</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">return_var_names</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwargs</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#scCCA.plots.cluster_enrichment.cluster_enrichment" title="Permalink to this definition"></a></dt>
<dd><p>Enrichment analysis of clusters based on differential gene expression.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>adata</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">AnnData</span></code>) – Anndata object.</p></li>
<li><p><strong>model_key</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">str</span></code>) – Key to access the model in the adata object.</p></li>
<li><p><strong>states</strong> (<code class="xref py py-data docutils literal notranslate"><span class="pre">Union</span></code>[<code class="xref py py-class docutils literal notranslate"><span class="pre">str</span></code>, <code class="xref py py-class docutils literal notranslate"><span class="pre">List</span></code>[<code class="xref py py-class docutils literal notranslate"><span class="pre">str</span></code>]]) – States for comparison. If not pair of states is provided, “Intercept” is assumed to be the base state.</p></li>
<li><p><strong>factor</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>) – Factor index to consider for the differential calculation.</p></li>
<li><p><strong>state_key</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">str</span></code>) – Key for state in adata.</p></li>
<li><p><strong>cluster_key</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">str</span></code>) – Key for cluster in adata.</p></li>
<li><p><strong>groups</strong> (<code class="xref py py-data docutils literal notranslate"><span class="pre">Union</span></code>[<code class="xref py py-class docutils literal notranslate"><span class="pre">str</span></code>, <code class="xref py py-class docutils literal notranslate"><span class="pre">List</span></code>[<code class="xref py py-class docutils literal notranslate"><span class="pre">str</span></code>]]) – Groups to consider for enrichment.</p></li>
<li><p><strong>dot</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>) – If True, a dot plot is generated. Otherwise, a heatmap is generated. Default is False.</p></li>
<li><p><strong>sign</strong> (<code class="xref py py-data docutils literal notranslate"><span class="pre">Union</span></code>[<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>, <code class="xref py py-class docutils literal notranslate"><span class="pre">float</span></code>]) – Sign to adjust the difference, by default 1.</p></li>
<li><p><strong>highest</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>) – Number of highest differential genes to retrieve, by default 0.</p></li>
<li><p><strong>lowest</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>) – Number of lowest differential genes to retrieve, by default 0.</p></li>
<li><p><strong>threshold</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">float</span></code>) – Threshold for significance, by default 1.96.</p></li>
<li><p><strong>return_var_names</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>) – If True, returns variable names. Otherwise, returns the plot axes. Default is False.</p></li>
<li><p><strong>**kwargs</strong> – Additional keyword arguments passed to the plotting function.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>If <cite>return_var_names</cite> is True, returns a dictionary of variable names. Otherwise, returns the plot axes.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>Union[dict, matplotlib.axes.Axes]</p>
</dd>
</dl>
<p class="rubric">Notes</p>
<p>This function performs enrichment analysis of clusters based on differential gene expression.
It first validates the sign and retrieves differential genes. Depending on the <cite>dot</cite> parameter, either a
dot plot or a heatmap is generated to visualize the enrichment.</p>
</dd></dl>

<span class="target" id="module-scCCA.plots.factor_embedding"></span><dl class="py function">
<dt class="sig sig-object py" id="scCCA.plots.factor_embedding.factor_embedding">
<span class="sig-prename descclassname"><span class="pre">scCCA.plots.factor_embedding.</span></span><span class="sig-name descname"><span class="pre">factor_embedding</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">adata</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">model_key</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">factor=None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">basis=None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sign=1.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cmap=&lt;matplotlib.colors.LinearSegmentedColormap</span> <span class="pre">object&gt;</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">colorbar_pos='right'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">colorbar_width='3%'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">orientation='vertical'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pad=0.1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">size=1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ncols=4</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">width=4</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">height=3</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ax=None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#scCCA.plots.factor_embedding.factor_embedding" title="Permalink to this definition"></a></dt>
<dd><p>Plot factor on a given basis.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>adata</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">AnnData</span></code>) – AnnData object.</p></li>
<li><p><strong>model_key</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">str</span></code>) – Key for the fitted model.</p></li>
<li><p><strong>basis</strong> (<code class="xref py py-data docutils literal notranslate"><span class="pre">Optional</span></code>[<code class="xref py py-class docutils literal notranslate"><span class="pre">str</span></code>]) – Key for the basis (e.g. UMAP, T-SNE). If basis is None factor embedding
tries to retrieve “X_{model_key}_umap”.</p></li>
<li><p><strong>factor</strong> (<code class="xref py py-data docutils literal notranslate"><span class="pre">Union</span></code>[<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>, <code class="xref py py-class docutils literal notranslate"><span class="pre">List</span></code>[<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>], <code class="xref py py-obj docutils literal notranslate"><span class="pre">None</span></code>]) – Factor(s) to plot. If None, then all factors are plotted.</p></li>
<li><p><strong>sign</strong> (<code class="xref py py-data docutils literal notranslate"><span class="pre">Union</span></code>[<code class="xref py py-class docutils literal notranslate"><span class="pre">float</span></code>, <code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>]) – Sign of the factor. Should be either 1.0 or -1.0.</p></li>
<li><p><strong>cmap</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">Colormap</span></code>) – Colormap for the scatterplot.</p></li>
<li><p><strong>colorbar_pos</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">str</span></code>) – Position of the colorbar.</p></li>
<li><p><strong>colorbar_width</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">str</span></code>) – Width of the colorbar.</p></li>
<li><p><strong>orientation</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">str</span></code>) – Orientation of the colorbar. Should be either “vertical” or “horizontal”.</p></li>
<li><p><strong>size</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">float</span></code>) – Marker/Dot size of the scatterplot.</p></li>
<li><p><strong>ncols</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>) – Number of columns for the subplots.</p></li>
<li><p><strong>width</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>) – Width of each subplot.</p></li>
<li><p><strong>height</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>) – Height of each subplot.</p></li>
<li><p><strong>ax</strong> (<code class="xref py py-data docutils literal notranslate"><span class="pre">Optional</span></code>[<code class="xref py py-class docutils literal notranslate"><span class="pre">Axes</span></code>]) – Axes object to plot on. If None, then a new figure is created. Works only
if one factor is plotted.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>Axes object.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>ax</p>
</dd>
</dl>
</dd></dl>

<span class="target" id="module-scCCA.plots.factor_enrichment"></span></section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="models.html" class="btn btn-neutral float-left" title="Models" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="notebooks/PBMC5K.html" class="btn btn-neutral float-right" title="Analysing PBMC5K using scCCA" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2022, Harald Vohringer.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>