<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Analysing PBMC5K using scCCA &mdash; scCCA 0.1.0 documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
        <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
        <script async="async" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
        <script>window.MathJax = {"tex": {"inlineMath": [["$", "$"], ["\\(", "\\)"]], "processEscapes": true}, "options": {"ignoreHtmlClass": "tex2jax_ignore|mathjax_ignore|document", "processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Analysing interferon-\(\beta\) stimulated PBMCs using scPCA" href="kang.html" />
    <link rel="prev" title="Plotting" href="../plots.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="../index.html" class="icon icon-home"> scCCA
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">API:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../models.html">Models</a></li>
<li class="toctree-l1"><a class="reference internal" href="../plots.html">Plotting</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Tutorials:</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">Analysing PBMC5K using scCCA</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#Preprocessing">Preprocessing</a></li>
<li class="toctree-l2"><a class="reference internal" href="#HVG-selection">HVG selection</a></li>
<li class="toctree-l2"><a class="reference internal" href="#Model-training">Model training</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#Model-fit">Model fit</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#Down-stream-analysis">Down stream analysis</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="kang.html">Analysing interferon-<span class="math notranslate nohighlight">\(\beta\)</span> stimulated PBMCs using scPCA</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">scCCA</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
      <li>Analysing PBMC5K using scCCA</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/notebooks/PBMC5K.ipynb.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  
<style>
/* CSS for nbsphinx extension */

/* remove conflicting styling from Sphinx themes */
div.nbinput.container div.prompt *,
div.nboutput.container div.prompt *,
div.nbinput.container div.input_area pre,
div.nboutput.container div.output_area pre,
div.nbinput.container div.input_area .highlight,
div.nboutput.container div.output_area .highlight {
    border: none;
    padding: 0;
    margin: 0;
    box-shadow: none;
}

div.nbinput.container > div[class*=highlight],
div.nboutput.container > div[class*=highlight] {
    margin: 0;
}

div.nbinput.container div.prompt *,
div.nboutput.container div.prompt * {
    background: none;
}

div.nboutput.container div.output_area .highlight,
div.nboutput.container div.output_area pre {
    background: unset;
}

div.nboutput.container div.output_area div.highlight {
    color: unset;  /* override Pygments text color */
}

/* avoid gaps between output lines */
div.nboutput.container div[class*=highlight] pre {
    line-height: normal;
}

/* input/output containers */
div.nbinput.container,
div.nboutput.container {
    display: -webkit-flex;
    display: flex;
    align-items: flex-start;
    margin: 0;
    width: 100%;
}
@media (max-width: 540px) {
    div.nbinput.container,
    div.nboutput.container {
        flex-direction: column;
    }
}

/* input container */
div.nbinput.container {
    padding-top: 5px;
}

/* last container */
div.nblast.container {
    padding-bottom: 5px;
}

/* input prompt */
div.nbinput.container div.prompt pre {
    color: #307FC1;
}

/* output prompt */
div.nboutput.container div.prompt pre {
    color: #BF5B3D;
}

/* all prompts */
div.nbinput.container div.prompt,
div.nboutput.container div.prompt {
    width: 4.5ex;
    padding-top: 5px;
    position: relative;
    user-select: none;
}

div.nbinput.container div.prompt > div,
div.nboutput.container div.prompt > div {
    position: absolute;
    right: 0;
    margin-right: 0.3ex;
}

@media (max-width: 540px) {
    div.nbinput.container div.prompt,
    div.nboutput.container div.prompt {
        width: unset;
        text-align: left;
        padding: 0.4em;
    }
    div.nboutput.container div.prompt.empty {
        padding: 0;
    }

    div.nbinput.container div.prompt > div,
    div.nboutput.container div.prompt > div {
        position: unset;
    }
}

/* disable scrollbars on prompts */
div.nbinput.container div.prompt pre,
div.nboutput.container div.prompt pre {
    overflow: hidden;
}

/* input/output area */
div.nbinput.container div.input_area,
div.nboutput.container div.output_area {
    -webkit-flex: 1;
    flex: 1;
    overflow: auto;
}
@media (max-width: 540px) {
    div.nbinput.container div.input_area,
    div.nboutput.container div.output_area {
        width: 100%;
    }
}

/* input area */
div.nbinput.container div.input_area {
    border: 1px solid #e0e0e0;
    border-radius: 2px;
    /*background: #f5f5f5;*/
}

/* override MathJax center alignment in output cells */
div.nboutput.container div[class*=MathJax] {
    text-align: left !important;
}

/* override sphinx.ext.imgmath center alignment in output cells */
div.nboutput.container div.math p {
    text-align: left;
}

/* standard error */
div.nboutput.container div.output_area.stderr {
    background: #fdd;
}

/* ANSI colors */
.ansi-black-fg { color: #3E424D; }
.ansi-black-bg { background-color: #3E424D; }
.ansi-black-intense-fg { color: #282C36; }
.ansi-black-intense-bg { background-color: #282C36; }
.ansi-red-fg { color: #E75C58; }
.ansi-red-bg { background-color: #E75C58; }
.ansi-red-intense-fg { color: #B22B31; }
.ansi-red-intense-bg { background-color: #B22B31; }
.ansi-green-fg { color: #00A250; }
.ansi-green-bg { background-color: #00A250; }
.ansi-green-intense-fg { color: #007427; }
.ansi-green-intense-bg { background-color: #007427; }
.ansi-yellow-fg { color: #DDB62B; }
.ansi-yellow-bg { background-color: #DDB62B; }
.ansi-yellow-intense-fg { color: #B27D12; }
.ansi-yellow-intense-bg { background-color: #B27D12; }
.ansi-blue-fg { color: #208FFB; }
.ansi-blue-bg { background-color: #208FFB; }
.ansi-blue-intense-fg { color: #0065CA; }
.ansi-blue-intense-bg { background-color: #0065CA; }
.ansi-magenta-fg { color: #D160C4; }
.ansi-magenta-bg { background-color: #D160C4; }
.ansi-magenta-intense-fg { color: #A03196; }
.ansi-magenta-intense-bg { background-color: #A03196; }
.ansi-cyan-fg { color: #60C6C8; }
.ansi-cyan-bg { background-color: #60C6C8; }
.ansi-cyan-intense-fg { color: #258F8F; }
.ansi-cyan-intense-bg { background-color: #258F8F; }
.ansi-white-fg { color: #C5C1B4; }
.ansi-white-bg { background-color: #C5C1B4; }
.ansi-white-intense-fg { color: #A1A6B2; }
.ansi-white-intense-bg { background-color: #A1A6B2; }

.ansi-default-inverse-fg { color: #FFFFFF; }
.ansi-default-inverse-bg { background-color: #000000; }

.ansi-bold { font-weight: bold; }
.ansi-underline { text-decoration: underline; }


div.nbinput.container div.input_area div[class*=highlight] > pre,
div.nboutput.container div.output_area div[class*=highlight] > pre,
div.nboutput.container div.output_area div[class*=highlight].math,
div.nboutput.container div.output_area.rendered_html,
div.nboutput.container div.output_area > div.output_javascript,
div.nboutput.container div.output_area:not(.rendered_html) > img{
    padding: 5px;
    margin: 0;
}

/* fix copybtn overflow problem in chromium (needed for 'sphinx_copybutton') */
div.nbinput.container div.input_area > div[class^='highlight'],
div.nboutput.container div.output_area > div[class^='highlight']{
    overflow-y: hidden;
}

/* hide copybtn icon on prompts (needed for 'sphinx_copybutton') */
.prompt .copybtn {
    display: none;
}

/* Some additional styling taken form the Jupyter notebook CSS */
.jp-RenderedHTMLCommon table,
div.rendered_html table {
  border: none;
  border-collapse: collapse;
  border-spacing: 0;
  color: black;
  font-size: 12px;
  table-layout: fixed;
}
.jp-RenderedHTMLCommon thead,
div.rendered_html thead {
  border-bottom: 1px solid black;
  vertical-align: bottom;
}
.jp-RenderedHTMLCommon tr,
.jp-RenderedHTMLCommon th,
.jp-RenderedHTMLCommon td,
div.rendered_html tr,
div.rendered_html th,
div.rendered_html td {
  text-align: right;
  vertical-align: middle;
  padding: 0.5em 0.5em;
  line-height: normal;
  white-space: normal;
  max-width: none;
  border: none;
}
.jp-RenderedHTMLCommon th,
div.rendered_html th {
  font-weight: bold;
}
.jp-RenderedHTMLCommon tbody tr:nth-child(odd),
div.rendered_html tbody tr:nth-child(odd) {
  background: #f5f5f5;
}
.jp-RenderedHTMLCommon tbody tr:hover,
div.rendered_html tbody tr:hover {
  background: rgba(66, 165, 245, 0.2);
}

/* CSS overrides for sphinx_rtd_theme */

/* 24px margin */
.nbinput.nblast.container,
.nboutput.nblast.container {
    margin-bottom: 19px;  /* padding has already 5px */
}

/* ... except between code cells! */
.nblast.container + .nbinput.container {
    margin-top: -19px;
}

.admonition > p:before {
    margin-right: 4px;  /* make room for the exclamation icon */
}

/* Fix math alignment, see https://github.com/rtfd/sphinx_rtd_theme/pull/686 */
.math {
    text-align: unset;
}
</style>
<p><a class="reference external" href="https://colab.research.google.com/github/sagar87/scCCA/blob/main/docs/notebooks/PBMC5K.ipynb"><img alt="Open In Colab" src="https://colab.research.google.com/assets/colab-badge.svg" /></a></p>
<p><em>In order to run this notebook in Google Colab uncomment and run the following cells. For best performance make sure that you have run the notebook on a GPU instance, i.e. choose from the menu bar Runtime &gt; Change Runtime &gt; Hardware accelerator &gt; GPU.</em></p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span># Install scCCA + depenencies
# !pip install --quiet scCCA
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span># Download the Kang et. al. dataset
# !wget https://www.huber.embl.de/users/harald/sc/5k_pbmc_protein_v3_filtered_feature_bc_matrix.h5 /content/5k_pbmc_protein_v3_filtered_feature_bc_matrix.h5
</pre></div>
</div>
</div>
<section id="Analysing-PBMC5K-using-scCCA">
<h1>Analysing PBMC5K using scCCA<a class="headerlink" href="#Analysing-PBMC5K-using-scCCA" title="Permalink to this headline"></a></h1>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[2]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>import pandas as pd
import scanpy as sc
import scCCA as scc
</pre></div>
</div>
</div>
<section id="Preprocessing">
<h2>Preprocessing<a class="headerlink" href="#Preprocessing" title="Permalink to this headline"></a></h2>
<p>In this tutorial we illustrate how to apply <code class="docutils literal notranslate"><span class="pre">scCCA</span></code> to the 10X PBMC5K dataset. We start by loading the the data using <code class="docutils literal notranslate"><span class="pre">scanpy</span></code>. Note that we have to pass the <code class="docutils literal notranslate"><span class="pre">gex_only=False</span></code> argument to avoid dropping the ADT counts.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[3]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>adata = sc.read_10x_h5(&#39;/content/5k_pbmc_protein_v3_filtered_feature_bc_matrix.h5&#39;, gex_only=False)
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
/home/voehring/.local/lib/python3.9/site-packages/anndata/_core/anndata.py:1830: UserWarning: Variable names are not unique. To make them unique, call `.var_names_make_unique`.
  utils.warn_names_duplicates(&#34;var&#34;)
</pre></div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[4]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>adata.var_names_make_unique()
</pre></div>
</div>
</div>
<p>Now, we separate the protein data from the RNA count matrix and store it separately in the <code class="docutils literal notranslate"><span class="pre">adata.obsm</span></code> field. The following lines of code extract the protein names and barcodes to create a <code class="docutils literal notranslate"><span class="pre">pandas.DataFrame</span></code> for the protein count data.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[5]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>col_names = adata.var[adata.var.feature_types == &#39;Antibody Capture&#39;].gene_ids.tolist()
row_names = adata.obs.index.tolist()
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[6]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>protein_counts = adata[:, adata.var.feature_types == &#39;Antibody Capture&#39;].X.toarray()
</pre></div>
</div>
</div>
<p>Before we save the protein count table to the <code class="docutils literal notranslate"><span class="pre">adata</span></code> object, we drop the control ADTs IgG1, IgG2 and IgG2b. Finally, we subset the <code class="docutils literal notranslate"><span class="pre">adata</span></code> object such that <code class="docutils literal notranslate"><span class="pre">adata.X</span></code> contains RNA counts only, while the ADT data is stored in <code class="docutils literal notranslate"><span class="pre">adata.obsm['protein_expression']</span></code>.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[7]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>adata.obsm[&#39;protein_expression&#39;] = pd.DataFrame(protein_counts, index=row_names, columns=col_names).drop(columns=[&#39;IgG1&#39;, &#39;IgG2a&#39;, &#39;IgG2b&#39;])
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[8]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>adata = adata[:, adata.var.feature_types == &#39;Gene Expression&#39;]
</pre></div>
</div>
</div>
</section>
<section id="HVG-selection">
<h2>HVG selection<a class="headerlink" href="#HVG-selection" title="Permalink to this headline"></a></h2>
<p>Next, we select highly variable genes from the data. We employ the <code class="docutils literal notranslate"><span class="pre">highly_variable_genes</span></code> function from <code class="docutils literal notranslate"><span class="pre">scanpy</span></code> using the <code class="docutils literal notranslate"><span class="pre">seurat_v3</span></code> settings which directly operates on count data (that is in contrast to other HVG selection procedures that require some sort of preprocessing). Supplying the <code class="docutils literal notranslate"><span class="pre">subset=True</span></code> argument drops all unneccessary features from the data such that we can directly proceed by fitting the <code class="docutils literal notranslate"><span class="pre">scCCA</span></code> model.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[9]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>sc.pp.highly_variable_genes(
    adata,
    n_top_genes=2000,
    flavor=&quot;seurat_v3&quot;,
    subset=True,
)
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
/home/voehring/.local/lib/python3.9/site-packages/anndata/compat/_overloaded_dict.py:106: ImplicitModificationWarning: Trying to modify attribute `._uns` of view, initializing view as actual.
  self.data[key] = value
</pre></div></div>
</div>
</section>
<section id="Model-training">
<h2>Model training<a class="headerlink" href="#Model-training" title="Permalink to this headline"></a></h2>
<p>Training the <code class="docutils literal notranslate"><span class="pre">scCCA</span></code> model is easy for this simple data set. All we need to provide is the <code class="docutils literal notranslate"><span class="pre">adata</span></code> object, the number of factors we want to fit and the respective <code class="docutils literal notranslate"><span class="pre">obsm</span></code> key for the protein data. The <code class="docutils literal notranslate"><span class="pre">scCCA</span></code> constructor returns a <code class="docutils literal notranslate"><span class="pre">scCCA</span></code> model instance which we can train by calling its <code class="docutils literal notranslate"><span class="pre">fit</span></code> method.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[10]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>m0 = scc.scCCA(adata, num_factors=15, protein_obsm_key=&#39;protein_expression&#39;)
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[11]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>m0.fit()
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
  0%|          | 0/5000 [00:00&lt;?, ?it/s]/home/voehring/voehring/opt/scCCA/scCCA/model.py:144: UserWarning: The use of `x.T` on tensors of dimension other than 2 to reverse their shape is deprecated and it will throw an error in a future release. Consider `x.mT` to transpose batches of matricesor `x.permute(*torch.arange(x.ndim - 1, -1, -1))` to reverse the dimensions of a tensor. (Triggered internally at  ../aten/src/ATen/native/TensorShape.cpp:2985.)
  α_rna = deterministic(&#34;α_rna&#34;, (1 / α_rna_inv).T)
Epoch: 4990 | lr: 1.00E-02 | ELBO: 4216514 | Δ_10: 11731.50 | Best: 4164478: 100%|██████████| 5000/5000 [06:14&lt;00:00, 13.35it/s]
</pre></div></div>
</div>
<p>In many cases it is adviseable to train the model for a few additional epochs with a smaller learning rate. We can call the <code class="docutils literal notranslate"><span class="pre">fit</span></code> method for a second time and pass the arguments <code class="docutils literal notranslate"><span class="pre">lr=0.001</span></code> to continue the training with a learning rate of 0.001 (initally it was 0.01).</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[12]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>m0.fit(lr=0.001)
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
Epoch: 9990 | lr: 1.00E-03 | ELBO: 4176325 | Δ_10: 14536.00 | Best: 4141708: 100%|██████████| 5000/5000 [05:49&lt;00:00, 14.30it/s]
</pre></div></div>
</div>
<p>To obtain the posterior mean of the fitted model parameters we call the <code class="docutils literal notranslate"><span class="pre">to_anndata</span></code> function and pass a sensible <code class="docutils literal notranslate"><span class="pre">model_key</span></code> argument. The <code class="docutils literal notranslate"><span class="pre">scCCA</span></code> class will save the loadings and factors in <code class="docutils literal notranslate"><span class="pre">adata.varm[model_key]</span></code> and <code class="docutils literal notranslate"><span class="pre">adata.obsm[f'X_{model_key}']</span></code>, respectively. Additional metadata and parameters are saved to <code class="docutils literal notranslate"><span class="pre">adata.uns[model_key]</span></code>.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[13]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>m0.to_anndata(model_key=&#39;rank15&#39;)
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
Predicting μ_rna for obs 4096-5246.: 100%|██████████| 3/3 [00:01&lt;00:00,  1.52it/s]
Predicting μ_prot for obs 4096-5246.: 100%|██████████| 3/3 [00:00&lt;00:00,  3.75it/s]
Predicting z for obs 4096-5246.: 100%|██████████| 3/3 [00:00&lt;00:00,  3.79it/s]
</pre></div></div>
</div>
<section id="Model-fit">
<h3>Model fit<a class="headerlink" href="#Model-fit" title="Permalink to this headline"></a></h3>
<p>Before moving on with more elaborate downstream analyses, we assess the fit of the model by using the following plotting functions.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[14]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>scc.pl.qc_hist(adata, model_key=&#39;rank15&#39;)
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[14]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
&lt;AxesSubplot: title={&#39;center&#39;: &#39;RMSE 10.47&#39;}, xlabel=&#39;True count ($\\log_{10}(x+1)$ scaled)&#39;, ylabel=&#39;Predicted count ($\\log_{10}(x+1)$ scaled)&#39;&gt;
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/notebooks_PBMC5K_25_1.png" src="../_images/notebooks_PBMC5K_25_1.png" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[16]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>scc.pl.qc_hist(adata, model_key=&#39;rank15&#39;, protein_obsm_key=&#39;protein_expression&#39;)
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[16]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
&lt;AxesSubplot: title={&#39;center&#39;: &#39;RMSE 25759.37&#39;}, xlabel=&#39;True count ($\\log_{10}(x+1)$ scaled)&#39;, ylabel=&#39;Predicted count ($\\log_{10}(x+1)$ scaled)&#39;&gt;
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/notebooks_PBMC5K_26_1.png" src="../_images/notebooks_PBMC5K_26_1.png" />
</div>
</div>
</section>
</section>
<section id="Down-stream-analysis">
<h2>Down stream analysis<a class="headerlink" href="#Down-stream-analysis" title="Permalink to this headline"></a></h2>
<p>Once we trust our model we can proceed with our prefered pipeline to analyse CITE-seq data.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[19]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>sc.pp.neighbors(adata, use_rep=&#39;X_rank15&#39;)
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[20]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>sc.tl.umap(adata)
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[24]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>sc.tl.leiden(adata)
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[25]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>sc.pl.umap(adata, color=&#39;leiden&#39;)
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
/home/voehring/.local/lib/python3.9/site-packages/scanpy/plotting/_tools/scatterplots.py:392: UserWarning: No data for colormapping provided via &#39;c&#39;. Parameters &#39;cmap&#39; will be ignored
  cax = scatter(
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/notebooks_PBMC5K_31_1.png" src="../_images/notebooks_PBMC5K_31_1.png" />
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[36]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>adata.obs = pd.concat([adata.obs, adata.obsm[&#39;protein_expression&#39;]], axis=1)
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[41]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>adata.obs
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[41]:
</pre></div>
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>leiden</th>
      <th>CD3</th>
      <th>CD4</th>
      <th>CD8a</th>
      <th>CD11b</th>
      <th>CD14</th>
      <th>CD15</th>
      <th>CD16</th>
      <th>CD19</th>
      <th>CD20</th>
      <th>...</th>
      <th>CD86</th>
      <th>CD127</th>
      <th>CD137</th>
      <th>CD197</th>
      <th>CD274</th>
      <th>CD278</th>
      <th>CD335</th>
      <th>PD-1</th>
      <th>HLA-DR</th>
      <th>TIGIT</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>AAACCCAAGAGACAAG-1</th>
      <td>1</td>
      <td>25.0</td>
      <td>164.0</td>
      <td>16.0</td>
      <td>3011.0</td>
      <td>696.0</td>
      <td>19.0</td>
      <td>10.0</td>
      <td>6.0</td>
      <td>10.0</td>
      <td>...</td>
      <td>317.0</td>
      <td>2.0</td>
      <td>8.0</td>
      <td>6.0</td>
      <td>1.0</td>
      <td>10.0</td>
      <td>4.0</td>
      <td>7.0</td>
      <td>573.0</td>
      <td>10.0</td>
    </tr>
    <tr>
      <th>AAACCCAAGGCCTAGA-1</th>
      <td>0</td>
      <td>959.0</td>
      <td>720.0</td>
      <td>8.0</td>
      <td>12.0</td>
      <td>12.0</td>
      <td>9.0</td>
      <td>4.0</td>
      <td>5.0</td>
      <td>6.0</td>
      <td>...</td>
      <td>2.0</td>
      <td>37.0</td>
      <td>0.0</td>
      <td>10.0</td>
      <td>3.0</td>
      <td>121.0</td>
      <td>5.0</td>
      <td>17.0</td>
      <td>15.0</td>
      <td>3.0</td>
    </tr>
    <tr>
      <th>AAACCCAGTCGTGCCA-1</th>
      <td>2</td>
      <td>942.0</td>
      <td>1647.0</td>
      <td>21.0</td>
      <td>11.0</td>
      <td>13.0</td>
      <td>8.0</td>
      <td>7.0</td>
      <td>0.0</td>
      <td>7.0</td>
      <td>...</td>
      <td>3.0</td>
      <td>117.0</td>
      <td>7.0</td>
      <td>15.0</td>
      <td>4.0</td>
      <td>92.0</td>
      <td>12.0</td>
      <td>39.0</td>
      <td>11.0</td>
      <td>3.0</td>
    </tr>
    <tr>
      <th>AAACCCATCGTGCATA-1</th>
      <td>2</td>
      <td>802.0</td>
      <td>1666.0</td>
      <td>5.0</td>
      <td>11.0</td>
      <td>9.0</td>
      <td>8.0</td>
      <td>3.0</td>
      <td>3.0</td>
      <td>9.0</td>
      <td>...</td>
      <td>6.0</td>
      <td>157.0</td>
      <td>3.0</td>
      <td>3.0</td>
      <td>5.0</td>
      <td>44.0</td>
      <td>6.0</td>
      <td>15.0</td>
      <td>19.0</td>
      <td>3.0</td>
    </tr>
    <tr>
      <th>AAACGAAAGACAAGCC-1</th>
      <td>12</td>
      <td>529.0</td>
      <td>542.0</td>
      <td>4.0</td>
      <td>1285.0</td>
      <td>311.0</td>
      <td>6.0</td>
      <td>2.0</td>
      <td>2.0</td>
      <td>6.0</td>
      <td>...</td>
      <td>47.0</td>
      <td>31.0</td>
      <td>4.0</td>
      <td>6.0</td>
      <td>0.0</td>
      <td>17.0</td>
      <td>1.0</td>
      <td>4.0</td>
      <td>42.0</td>
      <td>0.0</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>TTTGTTGCAGCACAAG-1</th>
      <td>6</td>
      <td>693.0</td>
      <td>9.0</td>
      <td>737.0</td>
      <td>20.0</td>
      <td>15.0</td>
      <td>6.0</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>14.0</td>
      <td>...</td>
      <td>4.0</td>
      <td>7.0</td>
      <td>1.0</td>
      <td>12.0</td>
      <td>3.0</td>
      <td>146.0</td>
      <td>5.0</td>
      <td>34.0</td>
      <td>23.0</td>
      <td>14.0</td>
    </tr>
    <tr>
      <th>TTTGTTGCAGTCTTCC-1</th>
      <td>2</td>
      <td>402.0</td>
      <td>1417.0</td>
      <td>8.0</td>
      <td>15.0</td>
      <td>9.0</td>
      <td>14.0</td>
      <td>2.0</td>
      <td>2.0</td>
      <td>5.0</td>
      <td>...</td>
      <td>3.0</td>
      <td>145.0</td>
      <td>1.0</td>
      <td>2.0</td>
      <td>1.0</td>
      <td>65.0</td>
      <td>8.0</td>
      <td>11.0</td>
      <td>6.0</td>
      <td>2.0</td>
    </tr>
    <tr>
      <th>TTTGTTGCATGGCCCA-1</th>
      <td>9</td>
      <td>401.0</td>
      <td>1.0</td>
      <td>222.0</td>
      <td>7.0</td>
      <td>17.0</td>
      <td>6.0</td>
      <td>1.0</td>
      <td>2.0</td>
      <td>9.0</td>
      <td>...</td>
      <td>0.0</td>
      <td>14.0</td>
      <td>0.0</td>
      <td>1.0</td>
      <td>2.0</td>
      <td>40.0</td>
      <td>0.0</td>
      <td>8.0</td>
      <td>40.0</td>
      <td>15.0</td>
    </tr>
    <tr>
      <th>TTTGTTGCATTGCCGG-1</th>
      <td>1</td>
      <td>6.0</td>
      <td>46.0</td>
      <td>3.0</td>
      <td>1027.0</td>
      <td>382.0</td>
      <td>1.0</td>
      <td>4.0</td>
      <td>4.0</td>
      <td>1.0</td>
      <td>...</td>
      <td>144.0</td>
      <td>0.0</td>
      <td>3.0</td>
      <td>7.0</td>
      <td>0.0</td>
      <td>3.0</td>
      <td>1.0</td>
      <td>3.0</td>
      <td>184.0</td>
      <td>1.0</td>
    </tr>
    <tr>
      <th>TTTGTTGGTCCGGCAT-1</th>
      <td>4</td>
      <td>1773.0</td>
      <td>1903.0</td>
      <td>9.0</td>
      <td>9.0</td>
      <td>8.0</td>
      <td>17.0</td>
      <td>1.0</td>
      <td>7.0</td>
      <td>22.0</td>
      <td>...</td>
      <td>4.0</td>
      <td>108.0</td>
      <td>1.0</td>
      <td>10.0</td>
      <td>6.0</td>
      <td>170.0</td>
      <td>4.0</td>
      <td>12.0</td>
      <td>32.0</td>
      <td>12.0</td>
    </tr>
  </tbody>
</table>
<p>5247 rows × 30 columns</p>
</div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[44]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>sc.pl.umap(adata, color=[&#39;CD4&#39;, &#39;CD8a&#39;, &#39;CD20&#39;, &#39;CD45RA&#39;, &#39;CD45RO&#39;])
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/notebooks_PBMC5K_34_0.png" src="../_images/notebooks_PBMC5K_34_0.png" />
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>
</pre></div>
</div>
</div>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="../plots.html" class="btn btn-neutral float-left" title="Plotting" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="kang.html" class="btn btn-neutral float-right" title="Analysing interferon-\(\beta\) stimulated PBMCs using scPCA" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2022, Harald Vohringer.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>